lib/ansible/plugins/connection/docker.py

```diff
@@ -74,15 +74,19 @@ def __init__(self, play_context, new_stdin, *args, **kwargs):
             if not self.docker_cmd:
                 raise AnsibleError("docker command not found in PATH")
 
-        self.can_copy_bothways = False
-
         docker_version = self._get_docker_version()
         if LooseVersion(docker_version) < LooseVersion('1.3'):
             raise AnsibleError('docker connection type requires docker 1.3 or higher')
-        # Docker cp in 1.8.0 sets the owner and group to root rather than the
-        # user that the docker container is set to use by default.
-        #if LooseVersion(docker_version) >= LooseVersion('1.8.0'):
-        #    self.can_copy_bothways = True
+
+        self.remote_user = None
+
+        if self._play_context.remote_user is not None:
+            if LooseVersion(docker_version) >= LooseVersion('1.7'):
+                # Support for specifying the exec user was added in docker 1.7
+                self.remote_user = self._play_context.remote_user
+            else:
+                # The default exec user is root, unless it was changed in the Dockerfile with USER
+                display.warning('docker {0} does not support remote_user, using container default'.format(docker_version))
 
```

Comment:
So we weren't going to issue a warning if _play_context.remote_user matches with what docker is going to use anyways.  So I suppose we probably want to do something like this in the else:
``` python
p = subprocess.Popen([self.docker_cmd, 'inspect', '--format', '{{.Config.Image}}', self._play_context.remote_addr], shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
out, err = p.communicate()
if p.returncode != 0:
    display.warning() # Warning message about being unable to get user to login as from  docker + out + err
docker_remote_user = out.strip() or 'root'
if docker_remote_user != self._play_context.remote_user:
  display.warning('docker {0} does not support remote_user, using container default'.format(docker_version))